src/services/mediaDevicePreferences.ts

```diff
@@ -0,0 +1,198 @@
+/**
+ * @copyright Copyright (c) 2024 Maksim Sukharev <antreesy.web@gmail.com>
+ *
+ * @author Maksim Sukharev <antreesy.web@gmail.com>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/**
+ * List all registered devices in order of their preferences
+ * Show whether device is currently unplugged or selected, if information is available
+ *
+ * @param devices list of available devices
+ * @param audioInputId id of currently selected audio input
+ * @param videoInputId id of currently selected video input
+ * @param audioInputList list of registered audio devices in order of preference
+ * @param videoInputList list of registered video devices in order of preference
+ */
+function listMediaDevices(devices: MediaDeviceInfo[], audioInputId: string, videoInputId: string, audioInputList: MediaDeviceInfo[], videoInputList: MediaDeviceInfo[]) {
+	const availableDevices = devices.map(device => device.deviceId).filter(id => id !== 'default')
+
+	const getDeviceString = (device: MediaDeviceInfo, index: number) => {
+		const isUnplugged = !availableDevices.includes(device.deviceId) ? ' (unplugged)' : ''
+		const isSelected = () => {
+			if (device.kind === 'audioinput') {
+				return device.deviceId === audioInputId ? ' (selected)' : ''
+			} else if (device.kind === 'videoinput') {
+				return device.deviceId === videoInputId ? ' (selected)' : ''
+			}
+		}
+		return `    ${index + 1}. ${device.label || device.deviceId}` + isUnplugged + isSelected()
+	}
+
+	// eslint-disable-next-line no-console
+	console.log(`Media devices:
+  Audio input:
+${audioInputList.map(getDeviceString).join('\n')}
+
+  Video input:
+${videoInputList.map(getDeviceString).join('\n')}
+`)
```

Comment:
Do not forget to remove the debug log :P

Reply:
That's needed to list devices in the console (so user can pass some information to admin, e.g)

Reply:
Then it should not be `log`